source/base/index_set.cc

```diff
@@ -520,6 +520,10 @@ IndexSet::add_indices(const IndexSet &other, const size_type offset)
 bool
 IndexSet::is_subset_of(const IndexSet &other) const
 {
+  Assert(size() == other.size(),
+         ExcMessage("One index set can only be a subset of another if they "
+                    "describe index spaces of the same size."));
```

Comment:
Could you print the numbers in the message? Similar to the case of `AssertDimension`?

Reply:
Sure!